python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -151,6 +152,15 @@ def lock(self):
         finally:
             root._lock.release()
 
+    @property
+    @contextmanager
+    def reg_lock(self):
+        self._reg_lock.acquire()
+        try:
+            yield
+        finally:
+            self._reg_lock.release()
+
```

Comment:
This contextmanager is not necessary as the threading.RLock will already work as a context:
[threading locks as contexts](https://docs.python.org/2/library/threading.html#using-locks-conditions-and-semaphores-in-the-with-statement)

this merely needs to return the lock, i.e.

```python
@property
def reg_lock(self):
    return self._reg_lock
```